arrow-array/src/ffi.rs

```diff
@@ -1514,4 +1515,40 @@ mod tests_from_ffi {
             &imported
         );
     }
+
+    fn roundtrip_list_array(array: ListArray) -> ListArray {
+        let data = array.into_data();
+
+        let array = FFI_ArrowArray::new(&data);
+        let schema = FFI_ArrowSchema::try_from(data.data_type()).unwrap();
+
+        let array = unsafe { from_ffi(array, &schema) }.unwrap();
+        ListArray::from(array)
+    }
+
+    #[test]
+    fn test_extend_imported_list_slice() {
+        let mut data = vec![];
+
+        for _i in 0..1000 {
+            let mut list = vec![];
+            for j in 0..100 {
+                list.push(Some(j));
+            }
+            data.push(Some(list));
+        }
```

Comment:
If all the sublists are identical, I fear the unit test wouldn't be effective at catching some kinds of errors. Maybe pull out `j` to the top level, so every value is unique?

```suggestion
        let mut j = 0;
        for _i in 0..1000 {
            let mut list = vec![];
            for _ in 0..100 {
                list.push(Some(j));
                j += 1;
            }
            data.push(Some(list));
        }
```

Reply:
Actually, the error would be happened during `extend_array` which is not related to the actual data. It is still a good idea to make each list unique. I tweaked the test a little bit. Thanks.